cli/datapipelines.go

```diff
@@ -0,0 +1,309 @@
+package cli
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"os"
+	"time"
+
+	"github.com/urfave/cli/v2"
+	"github.com/yosuke-furukawa/json5/encoding/json5"
+	"go.mongodb.org/mongo-driver/bson"
+	datapipelinespb "go.viam.com/api/app/datapipelines/v1"
+)
+
+type datapipelineListArgs struct {
+	OrgID string
+}
+
+// DatapipelineListAction lists all data pipelines for an organization.
+func DatapipelineListAction(c *cli.Context, args datapipelineListArgs) error {
+	client, err := newViamClient(c)
+	if err != nil {
+		return err
+	}
+
+	resp, err := client.datapipelinesClient.ListDataPipelines(context.Background(), &datapipelinespb.ListDataPipelinesRequest{
+		OrganizationId: args.OrgID,
+	})
+	if err != nil {
+		return err
+	}
+
+	for _, pipeline := range resp.GetDataPipelines() {
+		printf(c.App.Writer, "\t%s (ID: %s)", pipeline.Name, pipeline.Id)
+	}
+
+	return nil
+}
+
+type datapipelineCreateArgs struct {
+	OrgID    string
+	Name     string
+	Schedule string
+	MQL      string
+	MqlFile  string
+}
+
+// DatapipelineCreateAction creates a new data pipeline.
+func DatapipelineCreateAction(c *cli.Context, args datapipelineCreateArgs) error {
+	client, err := newViamClient(c)
+	if err != nil {
+		return err
+	}
+
+	mqlBinary, err := parseMQL(args.MQL, args.MqlFile)
+	if err != nil {
+		return err
+	}
+
+	resp, err := client.datapipelinesClient.CreateDataPipeline(context.Background(), &datapipelinespb.CreateDataPipelineRequest{
+		OrganizationId: args.OrgID,
+		Name:           args.Name,
+		Schedule:       args.Schedule,
+		MqlBinary:      mqlBinary,
+	})
+	if err != nil {
+		return fmt.Errorf("error creating data pipeline: %w", err)
+	}
+
+	printf(c.App.Writer, "%s (ID: %s) created.", args.Name, resp.GetId())
+
+	return nil
+}
+
+type datapipelineUpdateArgs struct {
+	ID       string
+	Name     string
+	Schedule string
+	MQL      string
+	MqlFile  string
+}
+
+// DatapipelineUpdateAction updates an existing data pipeline.
+func DatapipelineUpdateAction(c *cli.Context, args datapipelineUpdateArgs) error {
+	client, err := newViamClient(c)
+	if err != nil {
+		return err
+	}
+
+	resp, err := client.datapipelinesClient.GetDataPipeline(context.Background(), &datapipelinespb.GetDataPipelineRequest{
+		Id: args.ID,
+	})
+	if err != nil {
+		return fmt.Errorf("error getting data pipeline: %w", err)
+	}
+	current := resp.GetDataPipeline()
+
+	name := args.Name
+	if name == "" {
+		name = current.GetName()
+	}
+
+	schedule := args.Schedule
+	if schedule == "" {
+		schedule = current.GetSchedule()
+	}
+
+	mqlBinary := current.GetMqlBinary()
+	if args.MQL != "" || args.MqlFile != "" {
+		mqlBinary, err = parseMQL(args.MQL, args.MqlFile)
+		if err != nil {
+			return err
+		}
+	}
```

Comment:
The reason I originally had all fields required for `UpdateDataPipeline` was because it was impossible to determine if a user wanted to disable a pipeline or simply omitted that field in their request. However, now that we've split out the disable/enable functionality, we should probably move this logic to the endpoint? Since we're already retrieving the pipeline doc there. Doesn't need to change with this PR, but perhaps a future improvement

Reply:
tricker in gRPC than in CLI IMO! because in gRPC you're sending a full proto message so how to distinguish between omitting a field and sending a default value? https://google.aip.dev/161

Reply:
but what if the user wants to reset to default value? 🤔